### PR TITLE
feature: add bind mode

### PR DESCRIPTION
**Pull request by rudyfly**
### Ⅰ. Describe what this PR did
Add bind mode, it contains:
"ro/rw/dr/rr/z/Z/nocopy/private/rprivate/slave/rslave/shared/rshared"

common mode for all bind are: "ro/rw"，"z/Z" will be supported later,
"dr/rr/nocopy" just for volume, dr means direct replace, rr means random replace, nocopy will be supported later.
"private/rprivate/slave/rslave/shared/rshared" just for rootfs.

### Ⅳ. Describe how to verify it
see test/cli_volume_test.go:TestVolumeBindReplaceMode

### Ⅴ. Special notes for reviews

Signed-off-by: Rudy Zhang <rudyflyzhang@gmail.com>

### CODE REVIEW

File: apis/server/container_bridge.go [hunk 1]
```diff
@@ -309,6 +309,13 @@ func (s *Server) getContainer(ctx context.Context, rw http.ResponseWriter, req *
 		}
 	}
 
+	if len(meta.Mounts) > 0 {
```
Comment (allencloud) [inline apis/server/container_bridge.go]: I do not think we need this judgement, the range will ignore the situation of length 0 automatically.

File: daemon/mgr/container.go [hunk 2]
```diff
@@ -1290,3 +1348,49 @@ func checkBind(b string) ([]string, error) {
 
 	return arr, nil
 }
+
+func parseBindMode(mp *types.MountPoint, mode string) error {
```
Comment (allencloud) [inline daemon/mgr/container.go]: I am afraid that we need unit test for this function. 😄

File: apis/server/container_bridge.go (hunk 1 above)
Comment (rudyfly) [reply apis/server/container_bridge.go]: ok

Comment (rudyfly) [reply apis/server/container_bridge.go]: ok

File: daemon/mgr/spec_volume.go [hunk 3]
```diff
@@ -14,28 +13,43 @@ func setupMounts(ctx context.Context, c *ContainerMeta, spec *SpecWrapper) error
...
-			opt = append(opt, strings.Split(sd[2], ",")...)
-			// Set rootfs propagation, default setting is private.
-			if strings.Contains(sd[2], "rshared") {
-				s.Linux.RootfsPropagation = "rshared"
+
+		pg := mp.Propagation
+		rootfspg := s.Linux.RootfsPropagation
+		// Set rootfs propagation, default setting is private.
+		switch pg {
+		case "shared", "rshare":
```
Comment (YaoZengzeng) [inline daemon/mgr/spec_volume.go]: Only "shared" and "rshared" are valid?

File: daemon/mgr/container.go (hunk 2 above)
Comment (rudyfly) [reply daemon/mgr/container.go]: ok, I add it.

Comment (rudyfly) [reply daemon/mgr/container.go]: I have added unit tests. PTAL.

**Pull request merged**